karelics_vesc_can_driver/vesc_can_driver.py

```diff
@@ -189,12 +193,20 @@ def can_cb(self, msg: Frame):
             self.get_logger().info(str(e))
 
         # Publish the current status of the vescs in to ros world
-        # TODO Why do we do this for all the vescs? Should we process tick only for the current_vesc?
+        # TODO: Why do we do this for all the vescs? Should we process tick only for the current_vesc?
+        # TODO: talk with mart about this? is it for timestamping and synchronization across vescs?
```

Comment:
What do you mean with the current vescs?

The tick publishes the current values to all the vesc topics. If you only update one it will not publish on all the vesc topics and you can not see the status for all vescs

Reply:
This comment was originally added by me. As far as I understood, this can_cb callback is called every time when one of the vescs update. We are parsing vesc_id here and are doing update to a single vesc values (to which I referred as current vesc). But the tick is then processed for all the vescs, even for the ones which values are not updated.

Been a while since I've worked with vesc_can_driver, but if I'm right, this causes vescs to publish status with 800hz instead of expected 200hz, causing vesc to publish lots of status data that has not changed. We noticed the effects of this when calculating odometry, when checking the difference of how much the wheel has rotated between the ticks. We got lots of 0 values and speculated that this published duplicated information causes it.

We also wondered if was this done on purpose, for example to publish the status message for all the vescs at the same to synchronize timestamps across the status messages?

Reply:
Ah now I understand. Yes you are correct, this is a bug! It would better to only update the current vesc. Should not be to hard